fetch_data.py

```diff
@@ -419,25 +439,109 @@ def create_csv(packaged_data, fieldnames, package_coords, fill_di=None):
                 row_di[fieldname] = value
         # write the actual value
         row_di["value"] = get_from_dict(packaged_data, coords)
-        writer.writerow(row_di)
-    return output.getvalue()
+        rows.append(row_di)
+    return rows
 
 
-def return_csv(csv_data):
-    """Return the CSV data as a download
+def write_csv(csv_dicts, fieldnames, filename, metadata=None):
+    """
+    Creates and returns a downloadable CSV file from list of CSV dicts.
 
     Args:
-        csv_data (?): csv data created with create_csv() function
+        csv_dicts (list): CSV data created with build_csv_dicts function.
+        fieldnames (list): list of fields used to create CSV header row
+        filename (str): File name of downloaded CSV file
 
     Returns:
         CSV Response
     """
+    output = io.StringIO()
+    if metadata is not None:
+        output.write(metadata)
+    writer = csv.DictWriter(output, fieldnames=fieldnames)
+    writer.writeheader()
+    writer.writerows(csv_dicts)
     response = Response(
-        csv_data,
+        output.getvalue(),
         mimetype="text/csv",
         headers={
-            "Content-Type": 'text/csv; name="climate.csv"',
-            "Content-Disposition": 'attachment; filename="climate.csv"',
+            "Content-Type": 'text/csv; name="' + filename + '"',
+            "Content-Disposition": 'attachment; filename="' + filename + '"',
         },
     )
     return response
+
+
+def place_name(place_type, place_id):
```

Comment:
Sweet. It feels like the `validate_data.py` module might be a better fit for this? Fine to me as is though.

Reply:
I think you're right, `validate_data.py` is a better place for this. I'll get this fixed!

Reply:
Fixed in 6e2b6c3b4f9b8c5deab7212f69190a3b592f75ba.